src/_pytest/pathlib.py

```diff
@@ -365,3 +365,105 @@ def fnmatch_ex(pattern: str, path) -> bool:
 def parts(s: str) -> Set[str]:
     parts = s.split(sep)
     return {sep.join(parts[: i + 1]) or sep for i in range(len(parts))}
+
+
+def import_module(path, modname=None, ensuresyspath=True):
+    """ return path as an imported python module.
+
+    If modname is None, look for the containing package
+    and construct an according module name.
+    The module will be put/looked up in sys.modules.
+    if ensuresyspath is True then the root dir for importing
+    the file (taking __init__.py files into account) will
+    be prepended to sys.path if it isn't there already.
+    If ensuresyspath=="append" the root dir will be appended
+    if it isn't already contained in sys.path.
+    if ensuresyspath is False no modification of syspath happens.
+
+    Special value of ensuresyspath=="importlib" is intended
+    purely for using in pytest, it is capable only of importing
+    separate .py files outside packages, e.g. for test suite
+    without any __init__.py file. It effectively allows having
+    same-named test modules in different places and offers
+    mild opt-in via this option. Note that it works only in
+    recent versions of python.
+    """
+    import py.error
+
+    if not path.check():
+        raise py.error.ENOENT(path)
+
+    if ensuresyspath == "importlib":
+        import importlib.util
+
+        if modname is None:
+            modname = path.purebasename
+
+        for meta_importer in sys.meta_path:
+            spec = meta_importer.find_spec(modname, [path.dirname])
+            if spec is not None:
+                break
+        else:
+            spec = importlib.util.spec_from_file_location(modname, str(path))
+
+        if spec is None:
+            raise ImportError(
+                "Can't find module {} at location {}".format(modname, str(path))
+            )
+        mod = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(mod)
+        return mod
+
+    pkgpath = None
+    if modname is None:
+        pkgpath = path.pypkgpath()
+        if pkgpath is not None:
+            pkgroot = pkgpath.dirpath()
+            names = path.new(ext="").relto(pkgroot).split(path.sep)
+            if names[-1] == "__init__":
+                names.pop()
+            modname = ".".join(names)
+        else:
+            pkgroot = path.dirpath()
+            modname = path.purebasename
+
+        path._ensuresyspath(ensuresyspath, pkgroot)
+        __import__(modname)
+        mod = sys.modules[modname]
+        if path.basename == "__init__.py":
+            return mod  # we don't check anything as we might
+            # be in a namespace package ... too icky to check
+        modfile = mod.__file__
+        if modfile[-4:] in (".pyc", ".pyo"):
+            modfile = modfile[:-1]
+        elif modfile.endswith("$py.class"):
```

Comment:
IIUC `$py.class` is a Jython thing, which we don't support (since it only supports Python 2), so can maybe remove this.

(If you prefer to keep this just a pure copy/paste that makes sense, can maybe make these changes later).